components/Typography/src/MDCFontScaler.m

```diff
@@ -0,0 +1,324 @@
+      };
+      _scalingCurve = scalingCurve;
+    } else if ([MDCTextStyleCaption isEqualToString:textStyle]) {
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @11,
+        UIContentSizeCategorySmall : @11,
+        UIContentSizeCategoryMedium : @11,
+        UIContentSizeCategoryLarge : @12,
+        UIContentSizeCategoryExtraLarge : @14,
+        UIContentSizeCategoryExtraExtraLarge : @16,
+        UIContentSizeCategoryExtraExtraExtraLarge : @18,
+        UIContentSizeCategoryAccessibilityMedium : @20,
+        UIContentSizeCategoryAccessibilityLarge : @22,
+        UIContentSizeCategoryAccessibilityExtraLarge : @24,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @26,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @28
+      };
+      _scalingCurve = scalingCurve;
+    } else if ([MDCTextStyleOverline isEqualToString:textStyle]) {
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @8,
+        UIContentSizeCategorySmall : @8,
+        UIContentSizeCategoryMedium : @9,
+        UIContentSizeCategoryLarge : @10,
+        UIContentSizeCategoryExtraLarge : @12,
+        UIContentSizeCategoryExtraExtraLarge : @14,
+        UIContentSizeCategoryExtraExtraExtraLarge : @16,
+        UIContentSizeCategoryAccessibilityMedium : @18,
+        UIContentSizeCategoryAccessibilityLarge : @20,
+        UIContentSizeCategoryAccessibilityExtraLarge : @22,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @24,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @26
+      };
+      _scalingCurve = scalingCurve;
+    } else {
+      // If nothing matches, return the metrics for MDCTextStyleBody1
+      _textStyle = [MDCTextStyleBody1 copy];
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @13,
+        UIContentSizeCategorySmall : @14,
+        UIContentSizeCategoryMedium : @15,
+        UIContentSizeCategoryLarge : @16,
+        UIContentSizeCategoryExtraLarge : @18,
+        UIContentSizeCategoryExtraExtraLarge : @20,
+        UIContentSizeCategoryExtraExtraExtraLarge : @22,
+        UIContentSizeCategoryAccessibilityMedium : @26,
+        UIContentSizeCategoryAccessibilityLarge : @30,
+        UIContentSizeCategoryAccessibilityExtraLarge : @34,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @38,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @42
+      };
+      _scalingCurve = scalingCurve;
+    }
+  }
+
+  return self;
+}
+
+- (UIFont *)scaledFontWithFont:(UIFont *)font {
+  // If it is available, query the preferredContentSizeCategory.
+  UIContentSizeCategory sizeCategory = GetCurrentSizeCategory();
+
+  UIFont *scaledFont;
+  if (font.mdc_scalingCurve) {
+    scaledFont = [font mdc_scaledFontForSizeCategory:sizeCategory];
```

Comment:
Just curious here, should `font` use the `_scalingCurve` to scale itself even if it already contains a `mdc_scalingCurve`?

Reply:
We are intentionally using the scaling curve from the MDCFontScaler so the following scenario works as expected:
-  I have set Headline1 to be ComicSans@40 and attached a curve.
-  I want Body1 to use the same font metrics in terms of fontFace, fontWeight, etc., but be sizing according the Body1 curve.
- I *can* use the Headline1 font as the 'original' font to base Body1 off of

But to your point, we *always* want to set the font scaling curve to the scaler's version.  We can eliminate the if statement.

Good question!

Reply:
It makes sense. The scenario should still work after removing if statement. Thanks for the clarification!